src/main/java/wooteco/subway/domain/Path.java

```diff
@@ -1,48 +1,54 @@
 package wooteco.subway.domain;
 
 import java.util.List;
+import java.util.stream.Collectors;
 import org.jgrapht.GraphPath;
 import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
-import org.jgrapht.graph.DefaultWeightedEdge;
 import org.jgrapht.graph.WeightedMultigraph;
 
 public class Path {
-    private final WeightedMultigraph<Station, DefaultWeightedEdge> graph;
+    private final WeightedMultigraph<Long, ShortestPathEdge> graph
+            = new WeightedMultigraph<>(ShortestPathEdge.class);
 
-    public Path() {
-        this.graph = new WeightedMultigraph<>(DefaultWeightedEdge.class);
-    }
-
-    public void addAllStations(List<Station> stations) {
+    public Path(List<Station> stations, List<Section> sections) {
         for (Station station : stations) {
-            graph.addVertex(station);
+            graph.addVertex(station.getId());
+        }
+        for (Section section : sections) {
+            graph.addEdge(section.getUpStationId(), section.getDownStationId(),
```

Comment:
객체 생성은 변수로 따로 분리해주는게 코드 읽기에 더 좋지 않을까요

```suggestion
            final var shortestPathEdge = new ShortestPathEdge(section.getLineId(), section.getDistance());
            graph.addEdge(section.getUpStationId(), section.getDownStationId(), shortestPathEdge);
```